test-data/unit/fine-grained.test

```diff
@@ -8670,6 +8670,36 @@ def f(x: int) -> None: ...
 from typing import Union
 def f(x: Union[int, str]) -> None: ...
 
+[targets c, b]
+[targets2 a]
+[out]
+==
+==
+
+[case testCheckReprocessedTargets-only_when_nocache]
+from b import B
+
+class C(B):
+    def meth(self) -> None:
+        from b import f
+        f()
+
+[file b.py]
+class B: ...
+def f() -> int: ...
+
+[file b.py.2]
+class A: ...
+class B(A): ...
+def f() -> int: ...
+
+[file b.py.3]
+class A: ...
+class B(A): ...
+def f() -> str: ...
+
+[targets b, __main__, __main__.C.meth, __main__, __main__.C.meth]
```

Comment:
Why is `__main__` processed twice?

Reply:
This is because of `<C> -> C`. After we first reprocess `C` we detect that MRO changed, so `<C>` is triggered. I agree this is suboptimal, but I don't think there is a simple way to avoid this.